taesch/lib/view_model/custom_widget/add_item_dialog_vm.dart

```diff
@@ -1,11 +1,14 @@
 import 'package:taesch/api/repository.dart';
 import 'package:taesch/model/error_case.dart';
 import 'package:taesch/model/product.dart';
+import 'package:taesch/model/tag.dart';
 
 class AddItemDialogVM {
   final String title = 'Add Item to Shopping List';
   final String textFormHint = 'Enter Item';
-  late Product temp;
+  final String tagFormHint = 'tags (comma separated)';
+  Product? temp;
+  List<Tag> tags = [];
```

Comment:
Prinzipiell ein super Ansatz solche Elemente zwischenzuspeichern. Ich weiß, es war nicht in dem Issue beschrieben, aber ich denke, dass doppelte Tags vermieden werden sollten. Hierfür muss man nicht sehr viel dazuimplementieren. 

![image](https://user-images.githubusercontent.com/58072297/204559277-87cc6154-3ba6-490c-af94-e6d63d8fa6bd.png)


Folgendes Vorgehen könnte ich mir vorstellen:

Diese Klasse bekommt ein Attribut mit dem Datentyp ![HashSet](https://api.dart.dev/be/175791/dart-collection/HashSet-class.html). Das verhindert Duplikate von Haus aus.

Beim Absenden wird dies dann in eine ![Liste](https://api.dart.dev/be/175791/dart-collection/HashSet-class.html#toList) umgewandelt und dem Produkt mitgegeben.

Reply:
ja das passt so ^^, im schlimmsten Fall, wenn jemand da Einwände haben sollte, können wir schauen, wie das strukturiert werden soll. Sowas wie ein LinkedHashSet oder so, jetzt aber irrelevant.